### DISCUSSION TITLE
Hiding the main program window/handling on_close with the new API

### DISCUSSION THREAD

**Discussion by 4JX** [emilk/egui#3885] [Q&A]
Updating my app to work with `0.25.0` and I noticed the removal of the `on_close_event` (using `eframe`).

I've tried calling `ctx.send_viewport_cmd(ViewportCommand::Visible(false));` which I assumed would be the equivalent of `frame.set_visible(false);` but it doesn't seem to do anything.
![image](https://github.com/emilk/egui/assets/79868816/6b7a8634-9f50-4302-bf11-c215d98878af)

Similarly for converting from on_close to the new API this prevents the window from being closed but doesn't hide the window afterwards
```rust
if ctx.input(|i| i.viewport().close_requested()) {
    ctx.send_viewport_cmd(ViewportCommand::CancelClose);
    ctx.send_viewport_cmd(ViewportCommand::Visible(false));
}
```

What am I missing here?

**Comment by Wicpar** [comment]
it works on windows for me, but it doesn't seem possible to show again, 
```rust
  ctx.send_viewport_cmd(egui::ViewportCommand::Visible(true));
```
is never consumed

**Comment by izolyomi** [comment]
I've just encountered this with egui 0.31.1 on Windows 10. On my system, handling
```rust
ctx.send_viewport_cmd(egui::ViewportCommand::Visible(false));
```
works just fine, but setting it back to true again is just always ignored, so the window is never shown again.
Is there a known and reasonable workaround without platform-specific calls?

**Comment by lifei** [comment]
Long story, see this https://github.com/emilk/egui/issues/3655
